docs/aws/audit/kubernetesmonitoring/rules/is_version_recent.mdx

```diff
@@ -24,6 +24,226 @@ HIPAA, AWSWAF, HITRUST, NISTCSF
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. Make sure you have the necessary permissions to access the EKS clusters.
+
+2. Once the AWS CLI is set up, you can list all the EKS clusters in your account using the following command:
+   ```
+   aws eks list-clusters
+   ```
+   This command will return a list of all the EKS clusters in your account.
+
+3. To get the version of Kubernetes running on a specific EKS cluster, use the following command:
+   ```
+   aws eks describe-cluster --name <cluster-name> --query "cluster.version"
+   ```
+   Replace `<cluster-name>` with the name of your EKS cluster. This command will return the version of Kubernetes running on the specified EKS cluster.
+
+4. To check the latest stable version of Kubernetes, you can visit the Kubernetes GitHub page or use the following command:
+   ```
+   curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt
+   ```
+   Compare the version obtained from the EKS cluster with the latest stable version. If they are not the same, it means the EKS cluster is not using the latest stable version of Kubernetes.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary Python libraries: You will need the boto3 library, which is the Amazon Web Services (AWS) SDK for Python. This allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, and others. You can install it using pip:
+
+   ```python
+   pip install boto3
+   ```
+
+2. Import the necessary libraries and establish a session with AWS:
+
+   ```python
+   import boto3
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='us-west-2'
+   )
+   ```
+
+3. Use the EKS client to list all the clusters and their versions:
+
+   ```python
+   eks = session.client('eks')
+   clusters = eks.list_clusters()['clusters']
+   for cluster in clusters:
+       cluster_info = eks.describe_cluster(name=cluster)['cluster']
+       print(f"Cluster: {cluster}, Version: {cluster_info['version']}")
+   ```
+
+4. Compare the version of each cluster with the latest stable version of Kubernetes. You can get the latest stable version from the Kubernetes GitHub page or from its official website. If the version of the cluster is not the latest, it means there is a misconfiguration. Note that this step requires manual intervention as the latest stable version needs to be updated manually in the script.
+
+   ```python
+   latest_stable_version = '1.21'  # Update this manually
+   for cluster in clusters:
+       cluster_info = eks.describe_cluster(name=cluster)['cluster']
+       if cluster_info['version'] < latest_stable_version:
+           print(f"Cluster: {cluster} is not using the latest stable version of Kubernetes")
+   ```
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>